src/auto/app_info.rs

```diff
@@ -80,15 +84,26 @@ impl AppInfo {
         }
     }
 
-    //#[cfg(any(feature = "v2_50", feature = "dox"))]
-    //pub fn launch_default_for_uri_async<'a, 'b, P: Into<Option<&'a Cancellable>>, Q: Into<Option<&'b /*Unimplemented*/AsyncReadyCallback>>, R: Into<Option</*Unimplemented*/Fundamental: Pointer>>>(uri: &str, launch_context: &AppLaunchContext, cancellable: P, callback: Q, user_data: R) {
-    //    unsafe { TODO: call ffi::g_app_info_launch_default_for_uri_async() }
-    //}
-
-    //#[cfg(any(feature = "v2_50", feature = "dox"))]
-    //pub fn launch_default_for_uri_finish<P: IsA</*Ignored*/AsyncResult>>(result: &P) -> Result<(), Error> {
-    //    unsafe { TODO: call ffi::g_app_info_launch_default_for_uri_finish() }
-    //}
+    #[cfg(any(feature = "v2_50", feature = "dox"))]
+    pub fn launch_default_for_uri_async<P: Fn(Result<(), Error>) + Send + Sync>(uri: &str, launch_context: &AppLaunchContext, callback: P) -> Cancellable {
+        let cancellable = Cancellable::new();
+        let user_data: Box<Box<Fn(Result<(), Error>) + Send + Sync>> = Box::new(Box::new(callback));
+        extern "C" fn launch_default_for_uri_async_trampoline(_source_object: *mut gobject_ffi::GObject, res: *mut ffi::GAsyncResult, user_data: glib_ffi::gpointer)
+        {
+            unsafe {
+                let mut error = ptr::null_mut();
+                let _ = ffi::g_app_info_launch_default_for_uri_finish(res, &mut error);
+                let result = if error.is_null() { Ok(()) } else { Err(from_glib_full(error)) };
+                let callback: &Box<Fn(Result<(), Error>) + Send + Sync> = transmute(user_data);
```

Comment:
clippy will warn about this one. First it could be &&Fn (that's easy to fix), second it could be a cast instead of transmute (that's very noisy and we didn't fix it for signals either).

Also I like that the trampoline is inside the function. We should do the same for signals (can you create a GitHub ticket about that?).

Reply:
Or rather, the signal trampolines should disappear the way they are and replaced by a Closure callback :)